File: src/test/java/photos/brooklyn/practice/codingexercises/java/leetcode/solutions/ShortestPathToAllKeysTest.java
package photos.brooklyn.practice.codingexercises.java.leetcode.solutions;

import org.junit.Test;

import static org.junit.Assert.*;

public class ShortestPathToAllKeysTest {
    private ShortestPathToAllKeys obj = new ShortestPathToAllKeys();

    @Test
    public void shortestPathAllKeys_1() {
        assertEquals(8, obj.shortestPathAllKeys(new String[]{"@.a.#","###.#","b.A.B"}));
    }

    @Test
    public void shortestPathAllKeys_2() {
        assertEquals(6, obj.shortestPathAllKeys(new String[]{"@..aA","..B#.","....b"}));
    }

    @Test
    public void shortestPathAllKeys_3() {
        assertEquals(7, obj.shortestPathAllKeys(new String[]{"@..a","##.#",".bA."}));
    }
}

File: test/scala/photos/brooklyn/practice/codingexercises/scala/leetcode/ShortestPathToAllKeysTest.scala
package photos.brooklyn.practice.codingexercises.scala.leetcode

import org.scalatest.{FlatSpec, Matchers}
import ShortestPathToAllKeys._

class ShortestPathToAllKeysTest extends FlatSpec with Matchers {

  behavior of "ShortestPathToAllKeysTest"

  it should "shortestPathAllKeys for test 1" in {
    shortestPathAllKeys(Array("@.a.#", "###.#", "b.A.B")) shouldEqual 8
  }
  it should "shortestPathAllKeys for test 2" in {
    shortestPathAllKeys(Array("@..aA", "..B#.", "....b")) shouldEqual 6
  }
  it should "shortestPathAllKeys for test 3" in {
    shortestPathAllKeys(Array("@..a", "##.#", ".bA.")) shouldEqual 7
  }
  it should "shortestPathAllKeys for test impossible" in {
    shortestPathAllKeys(Array("@..a", "####", ".bA.")) shouldEqual -1
  }
  it should "shortestPathAllKeys for test impossible 2" in {
    shortestPathAllKeys(Array("@Aa")) shouldEqual -1
  }

}
